MainUI.cs

```diff
@@ -60,16 +61,25 @@ private void LoadTransactionsOnUI(List<Transaction> list)
         {
             flowLayoutPanelTransactions.Controls.Clear();
             Console.WriteLine("search " + list.Count);
-            foreach (Transaction t in list)
+            if((list != null) && (list.First() != null))
             {
-                ListItemTransactions item = new ListItemTransactions
+                foreach (Transaction t in list)
                 {
-                    TransactionType = t.TransactionType,
-                    Title = t.Title,
-                    Amount = t.Amount.ToString()
-                };
-                flowLayoutPanelTransactions.Controls.Add(item);
+                    if (t == null)
+                    {
+                        continue;
+                    }
+                    ListItemTransactions item = new ListItemTransactions
```

Comment:
naming is a bit confusing. Is the object a list of transactions? Or is it a single transaction?
Judging by the contents used to create it, it looks like a single transaction, then it could just be named Transaction or something similar.